shared/post_metadata.go

```diff
@@ -8,6 +13,31 @@ type PostMetadata struct {
 	LabelsPerUnit uint64
 	NumUnits      uint32
 	MaxFileSize   uint64
-	Nonce         *uint64 `json:",omitempty"`
-	LastPosition  *uint64 `json:",omitempty"`
+	Nonce         *uint64    `json:",omitempty"`
+	NonceValue    NonceValue `json:",omitempty"`
+	LastPosition  *uint64    `json:",omitempty"`
+}
+
+type NonceValue []byte
+
+// UnmarshalJSON from hex
+func (n *NonceValue) UnmarshalJSON(data []byte) error {
+	if len(data) == 0 {
+		return nil
+	}
+	// trim quotes
+	data = data[1 : len(data)-1]
+	*n = make([]byte, hex.DecodedLen(len(data)))
+	_, err := hex.Decode(*n, data)
+	return err
+}
+
+// MarshalJSON to hex
+func (n NonceValue) MarshalJSON() ([]byte, error) {
+	if n == nil {
+		return []byte{}, nil
+	}
+	dst := make([]byte, hex.EncodedLen(len(n)))
+	hex.Encode(dst, n[:])
+	return []byte(fmt.Sprintf("\"%s\"", dst)), nil
 }
```

Comment:
No need to make it more complicated then necessary 😉 

```suggestion
type NonceValue []byte

func (n *NonceValue) MarshalJSON() ([]byte, error) {
	return json.Marshal(hex.EncodeToString(*n))
}

func (n *NonveValue) UnmarshalJSON(data []byte) (err error) {
	var hexString string
	if err = json.Unmarshal(data, &hexString); err != nil {
		return
	}
	*n, err = hex.DecodeString(hexString)
	return
}
```

Reply:
![](https://media3.giphy.com/media/RipfZWzjUDH25euMpM/giphy.gif?cid=ecf05e470kheeuxul89o9a4p5xw0ytmqytqflay8x476iknf&ep=v1_gifs_search&rid=giphy.gif&ct=g)